Pyrado/pyrado/algorithms/meta/sprl.py

```diff
@@ -0,0 +1,573 @@
+                                  and similar methods.
+        :param cov_is_parameter: if `True`, the covariance is treated as a parameter and returned from `get_stacked`
+                                 and similar methods.
+        """
+        super().__init__(mean, cov_chol_flat)
+
+        self._mean_is_parameter = mean_is_parameter
+        self._cov_is_parameter = cov_is_parameter
+
+    def from_stacked(self, stacked: np.ndarray) -> "ParameterAgnosticMultivariateNormalWrapper":
+        """
+        Builds a new `ParameterAgnosticMultivariateNormalWrapper` from the given stacked values. In contrast to
+        `MultivariateNormalWrapper.from_stacked(dim, stacked)`, this does not require a dimensionality as it is an
+        instance rather than a static method. Also, the stacked representations has to either contain the mean or the
+        standard deviations or both, according the the values originally passed to the constructor. If one of them is
+        not treated as a parameter, the current values is copied instead.
+
+        :param stacked: the stacked representation of the parameters according to the documentation above; can have
+                        either shape `(0,)`, `(k,)`, or `(2 * k)`
+        :return: a `ParameterAgnosticMultivariateNormalWrapper` with the new values for the parameters
+        """
+        if not (len(stacked.shape) == 1):
+            raise pyrado.ValueErr(msg="Stacked has invalid shape! Must be 1-dimensional.")
+
+        expected_dim_multiplier = 0
+        if self._mean_is_parameter:
+            expected_dim_multiplier += 1
+        if self._cov_is_parameter:
+            expected_dim_multiplier += 1
+        if not (stacked.shape[0] == expected_dim_multiplier * self.dim):
+            raise pyrado.ValueErr(msg=f"Stacked has invalid size! Must be {expected_dim_multiplier}*dim.")
+
+        if self._mean_is_parameter and self._cov_is_parameter:
+            mean = stacked[: self.dim]
+            cov_chol_flat = stacked[self.dim :]
+        elif self._mean_is_parameter and not self._cov_is_parameter:
+            mean = stacked[: self.dim]
+            cov_chol_flat = self.cov_chol_flat
+        elif not self._mean_is_parameter and self._cov_is_parameter:
+            mean = self.mean
+            cov_chol_flat = stacked
+        else:
+            mean = self.mean
+            cov_chol_flat = self.cov_chol_flat
+
+        if type(mean) == np.ndarray:
+            mean = to.tensor(mean).double()
+
+        if type(cov_chol_flat) == np.ndarray:
+            cov_chol_flat = to.tensor(cov_chol_flat).double()
+
+        return ParameterAgnosticMultivariateNormalWrapper(
+            mean=mean,
+            cov_chol_flat=cov_chol_flat,
+            mean_is_parameter=self._mean_is_parameter,
+            cov_is_parameter=self._cov_is_parameter,
+        )
+
+    def parameters(self) -> List[to.Tensor]:
```

Comment:
This function reminds me of torch.nn.Modules.parameters() which returns a generator. How about you do it the same way. As far as I see, you are only using it once later (actually to use it liek a generator). Establishing this parallelism would be nice.

Reply:
Done in bcdc32aa4a8d02ea06ca04b85098b97bb60c1447